Composer Additions

mikeselander opened the pull request:
Additions to the Composer file based on feedback in #79

tfrommen commented on composer.json:
```diff
@@ -1,7 +1,8 @@
 	"name": "hm-base",
-	"license": "GPL V3+",
+	"license": "GPL-3.0-or-later",
 	"minimum-stability": "stable",
```
So, what about this instead?

```json
    "minimum-stability": "dev",
    "prefer-stable": true,
```

mikeselander replied:
What is the benefit of this instead of removing that line and leaving it at `stable` by default?

tfrommen replied:
This would allow us to fetch non-stable versions when we need/want that. If there is a stable version according to our constraint, we will always get that. If there is not, then Composer will check for dev versions.

mikeselander replied:
Makes a lot of sense, thanks 👍

tfrommen commented on composer.json:
```diff
@@ -1,7 +1,8 @@
-	"license": "GPL V3+",
+	"license": "GPL-3.0-or-later",
 	"minimum-stability": "stable",
+	"type": "project",
```
Do we want to follow the _default_ ordering of the properties, or not?

I read that Paul thinks this is too obsessive, but isn't this all about consistency, readability and even discoverability?
Sure, it's nothing that we should spent too much time over, every now and then again. That's exactly why I would like to set this once, now, and why not use the official documentation (that other tools are based on, or that they are using internally themselves, for example, the normalize package)...?

The order of the current properties would then be like so:

```json
	"name": "hm-base",
	"type": "project",
	"license": "GPL-3.0-or-later",
	"require": { ... },
	"require-dev": { ... },
	"minimum-stability": "dev",
	"prefer-stable": true,
	"repositories": [ ... ],
	"extra": { ... },
```

And I would add the following right before `"extra"`:

```json
	"config": {
		"platform": {
			"php": "7.0"   // Or whatever our required minimum PHP version is.
		}
	}
```

Also, this is still missing something like `"php": ">=7.0"` in the `"require"` property (as first entry).

mikeselander commented:
@tfrommen  I have no preference on the ordering so I just moved them around to match that style since it seems like it's important to you.

I also added PHP, but at a more specific versioning constraint to ensure all devs are on the same major version. I expect this to get updated to 7.2 shortly anyhow.

mikeselander commented:
Also, I think this is ready for a final look-through unless anyone has more comments to add.

tfrommen commented:
>  I have no preference on the ordering so I just moved them around to match that style since it seems like it's important to you.

Well, it is not _important_ to me, I just thought that doing it in a _standardized_ way was (much more) preferable than doing it _just so_. 🤷‍♂️ 🙂
Thanks for ordering! 😅

The PHP versions are not yet right. I'll leave in-commit comments.

tfrommen commented on composer.json:
```diff
@@ -1,18 +1,26 @@
+	"type": "project",
+	"license": "GPL-3.0-or-later",
+	"require": {
+		"php": "7.0.*",
```
This would mean that you cannot install when you are running anything other than PHP 7.0.x locally. 7.1 and beyond wouldn't work, So, what we want here is this: `"php": ">=7.0"`, or, if we really want to have a specific patch release as minimum (which should almost never be the case), then we'd have something like `"php": ">=7.0.7"`.

![2018-09-14 10_13_33-mingw64__c_temp_composer-test](https://user-images.githubusercontent.com/6049306/45538287-f1bfe600-b806-11e8-887a-604a9900aed5.png)

tfrommen commented on composer.json:
```diff
@@ -1,18 +1,26 @@
-		"oomphinc/composer-installers-extender": "*"
+	"config": {
+		"platform": {
+			"php": "7.0"
```
Just a note: here we **do** want to be precise. So, **iff** we were to rely on a specific patch release of PHP, then this should go in here, for example, `"php": "7.0.7"`.

In our case, I think `7.0` is just fine (and then `7.2` sometime soon).

mikeselander replied:
I don't think that we can rely on a specific version of PHP for platform purposes on either VIP or our platform due to security updates. Neither platform informs developers of minor version bumps so `x.x` is as close as we can get for sure :/

tfrommen replied:
Ah, no. That's no problem. Whatever we put in the platform config is being used to generate the `.lock` file. No more. So we only have to make sure that all systems are running **at least** what we put in the platform config.
